Slim/Web/Time.pm

```diff
@@ -28,13 +31,29 @@ my $log = logger('network.http');
 
 # 'init' is called by 'Slim::Web::HTTP:init'
 sub init {
-	Slim::Web::Pages->addRawFunction(qr{^/tz$}, \&tzAPIrequest);
+	Slim::Web::Pages->addRawFunction(qr{^/time/tz$}, \&tzAPIrequest);
 }
 
+# Holds the Timezone string retrieved from a successful API call.
+# Although it may be '' if the string failed validation.
+my $cachedAPIresult;
 
 sub tzAPIrequest {
 	my ($httpClient, $response) = @_;
 
+	if (defined $cachedAPIresult) {
+		# Did it pass validation ? Or do we have '' ?
+		if ($cachedAPIresult) {
+			$log->info("TimeZone query: Returning \"$cachedAPIresult\" to SqueezeOS device (from cache)");
+			_sendHTTPresponse($httpClient, $response, RC_OK, $cachedAPIresult);
+		}
+		else {
+			$log->error("TimeZone query: Cached timezone was invalid. Returning 204 response to SqueezeOS device");
+			_sendHTTPresponse($httpClient, $response, RC_NO_CONTENT, '');
+		}
+		return;
+	}
```

Comment:
Minor detail: I'd remove the outer condition, as the first inner condition would imply a defined `$cachedAPIResult`. Therefore this could become just:

```perl
if ($cachedAPIresult) {
	$log->info();
	return _sendHTTPResponse();
}
elsif (defined $cachedAPIResult) {
	$log->warn();
	return _sendHTTPResponse();
}
```

And I'd change from `error()` to `warn()` because otherwise you'd get that message all the time. But the error already gets logged when we receive the value.

Reply:
I'm fine with not caching the failure state. Go for it. 

BTW: last night I was working on TODAL - which for some requests would accept time zone information... I guess I'll extend your work to provide the information internally via a CLI request or similar 😀.

Reply:
I still believe it's good to have a web endpoint to simplify changes on the player side. But it should be relatively easy to re-use your code to also provide that information over a CLI request. Thanks a lot for the initiative!

Reply:
I agree, it is much better result.

Thank you for your patient guidance...